app-example/src/main/java/com/badoo/ribs/example/app_bar/AppBarView.kt

```diff
@@ -0,0 +1,81 @@
+package com.badoo.ribs.example.app_bar
+
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.annotation.DrawableRes
+import androidx.annotation.LayoutRes
+import com.badoo.mvicore.ModelWatcher
+import com.badoo.mvicore.modelWatcher
+import com.badoo.ribs.core.customisation.inflate
+import com.badoo.ribs.core.view.RibView
+import com.badoo.ribs.core.view.ViewFactory
+import com.badoo.ribs.example.R
+import com.badoo.ribs.example.app_bar.AppBarView.Event
+import com.badoo.ribs.example.app_bar.AppBarView.ViewModel
+import com.badoo.ribs.example.image.ImageDownloader
+import com.badoo.ribs.example.view.TargettableImageView
+import com.jakewharton.rxrelay2.PublishRelay
+import io.reactivex.ObservableSource
+import io.reactivex.functions.Consumer
+
+interface AppBarView : RibView,
+    ObservableSource<Event>,
+    Consumer<ViewModel> {
+
+    sealed class Event {
+        object UserClicked : Event()
+        object SearchClicked : Event()
+    }
+
+    data class ViewModel(
+        val profileImageUrl: String,
+        @DrawableRes val placeholderRes: Int
```

Comment:
Maybe placeholder could be left a internal part of view impementation

Reply:
+1

Reply:
Yep, will do